cincinnati/src/lib.rs

```diff
@@ -237,14 +238,77 @@ impl Serialize for Graph {
     }
 }
 
+impl PartialEq for Graph {
+    fn eq(&self, other: &Graph) -> bool {
+        use daggy::petgraph::visit::IntoNeighbors;
+
+        let asc_order_release_by_version = {
+            use std::cmp::Ordering::{self, *};
+
+            |a: &&Release, b: &&Release| -> Ordering {
+                if a.version() < b.version() {
+                    Less
+                } else if a.version() == b.version() {
+                    Equal
+                } else {
+                    Greater
+                }
+            }
+        };
+
+        // Look through all nodes in self
+        self.dag.node_references().all(|node_ref| {
+            let dag_other = &other.dag;
+            let node_index = node_ref.0;
+            let release = node_ref.1;
+
+            // For each node in self, look through all nodes in other and find a
+            // match
+            dag_other
+                .node_references()
+                .filter(|node_ref_other| {
+                    let node_index_other = node_ref_other.0;
+                    let release_other = node_ref_other.1;
+
+                    // Node matches
+                    release == release_other && {
+                        let (mut neighbor_releases, mut neighbors_releases_other) = (vec![], vec![]);
+                        // Determine neighbors of own and matching node
+                        self.dag
+                            .neighbors(node_index)
+                            .zip(dag_other.neighbors(node_index_other))
+                            .for_each(|(neighbor, neighbor_other)| {
+                                let neighbor_release = self.dag.node_weight(neighbor).expect(EXPECT_NODE_WEIGHT);
+                                let neighbor_release_other = dag_other.node_weight(neighbor_other).expect(EXPECT_NODE_WEIGHT);
+                                neighbor_releases.push(neighbor_release);
+                                neighbors_releases_other.push(neighbor_release_other);
+                            });
+
+                        // Ensure the neighbors are equal by sorting and
+                        // comparing them one by one
+                        neighbor_releases.sort_by(asc_order_release_by_version);
+                        neighbors_releases_other.sort_by(asc_order_release_by_version);
+                        neighbor_releases
+                            .iter()
+                            .zip(neighbors_releases_other.iter())
+                            .all(|(neighbor, neighbor_other)| neighbor == neighbor_other)
+                    }
+                })
+                .count()
+                // Ensure each node in self has exactly one matching node in
+                // other , including its neighbors
+                == 1
+        })
+    }
+}
+
 #[cfg(test)]
 mod tests {
     extern crate serde_json;
 
     use super::*;
 
-    #[test]
-    fn serialize_graph() {
+    fn get_graph() -> Graph {
```

Comment:
Minor nit: I usually try to avoid `get_` as it is very generic verb. I would call something like this `generate_graph`.